ops/pebble.py

```diff
@@ -1966,3 +1952,162 @@ def send_signal(self, sig: typing.Union[int, str], services: typing.List[str]):
             'services': services,
         }
         self._request('POST', '/v1/signals', body=body)
+
+
+class MultipartLargeFileParser:
+    """A limited purpose multi-part parser backed by files for memory efficiency."""
+
+    def __init__(self, boundary: bytes):
+        self._response = None
+        self._files = {}
+
+        # RFC 2046 says that the boundary string needs to be preceded by a CRLF.
+        # Unfortunately, the request library's header parsing logic strips off one of
+        # these, so we'll prime the buffer with that missing sequence.
+        self._buffer = b'\r\n'
+
+        # Prepare the MIME multipart boundary line patterns.
+        # Note: RFC 2046 notes optional "linear whitespace" (e.g. [ \t]) after the boundary pattern
+        # and the optional "--" suffix.  Pebble doesn't use this, so to simplify implementation I
+        # am ignoring it.
+        self._boundary = "\r\n--{}\r\n".format(boundary).encode()
+        self._terminal_boundary = "\r\n--{}--\r\n".format(boundary).encode()
```

Comment:
I believe this is incorrect, as `'{}'.format(my_bytes)` will give include the `b'` quotes in the output. (It would probably still work, but not be what we want. Also, just concatenating the bytes avoids the call to `format` and `encode`:

```
>>> '\r\n--{}\r\n'.format(boundary).encode()
b"\r\n--b'0123456789abcdef'\r\n"
>>> b'\r\n--' + boundary + b'\r\n'
b'\r\n--0123456789abcdef\r\n'
```